lib/mpl_toolkits/axes_grid1/inset_locator.py

```diff
@@ -420,14 +435,27 @@ def inset_axes(parent_axes, width, height, loc=1,
             'center'       : 10
 
     bbox_to_anchor : tuple or `matplotlib.transforms.BboxBase`, optional
-        Bbox that the inset axes will be anchored. Can be a tuple of
-        [left, bottom, width, height], or a tuple of [left, bottom].
+        Bbox that the inset axes will be anchored to. If None,
+        *parent_axes.bbox* is used. If a tuple, can be either
+        [left, bottom, width, height], or [left, bottom].
+        If *width* and/or *height* are given in relative units, the 2-tuple
+        [left, bottom] cannot be used. Note that
+        the units of the bounding box are determined through the transform
+        in use. When using *bbox_to_anchor* it almost always makes sense to
+        also specify a *bbox_transform*. This might often be the axes transform
+        *parent_axes.transAxes*.
 
     bbox_transform : `matplotlib.transforms.Transform`, optional
-        Transformation for the bbox. if None, `parent_axes.transAxes` is used.
+        Transformation for the bbox. If None, a `.transforms.IdentityTransform`
+        is used. This is useful when not providing any argument to
+        *bbox_to_anchor*. When using *bbox_to_anchor* it almost always makes
```

Comment:
I don't understand whats useful about specifying "None", when *not* specifying a bbox_to_anchor?  In that case surely the bbox_to_anchor is the parent_axes and the transform is transAxes?

Reply:
That's the default case: `inset_axes(parent_axes, width="40%", height="30%")`. I suppose that is how people want to use this most often. In that case, `bbox_to_achor` is `parent_axes.bbox` and `bbox_transform` is `None`, which is the `transforms.IdentityTransform()`. This is *not* `transAxes`.

Reply:
Width and height get turned into pixels?

Reply:
Well, `width` and `height` are either in inches or relative to the bounding box (percentage). At the end of the day, *everything* is turned into pixels. 

Concerning the bounding box, if `bbox_transform` is `None`, the `IdentityTransform()` is used. That means that the `bbox_to_anchor` is in pixels. This makes sense, because the standard `parent_axes.bbox` is e.g. 

(x0=0.125, y0=0.11, x1=0.9, y1=0.88), transformed to (x0=0.0, y0=0.0, x1=6.4, y1=4.8), transformed by the dpi of 100, resulting in 

`(80.0, 52.8, 496.0, 369.6)` which is in pixels.

Reply:
I don't think *anything* should be thought of as pixels until draw time.  Axes bboxes are in figure-relative units (as they should be), and not converted until draw time to pixels.  Because at draw time, the dpi changes more often than not.  See all the inconsistencies in the docs and code that have to do w/ dpi.  I *really* don't think the `bbox_transform` should default to `None` here, but rather to the units consistent with the default for `bbox_to_anchor` which is figure-coordinates.  But I appreciate this was the historical way it was done, so we are stuck with it...

Reply:
Well, first of all Axes bboxes are `TransformedBbox`es. So they are somehow transient until draw time. But then they evaluate to pixels. This is not a problem at all. The problem only occurs when allowing the user to specify some other bounding box. Since the user usually won't create a `TransformedBbox`, but rather use a static tuple here, they would usually need to set a different transform as well. But we wouldn't want to include the complete logic of guessing what the user wants inside of this function, right? Like "if a bbox_to_anchor is specified and it is a tuple, chances are 98% that the user would have wanted to use the ax.transAxes, so let's change this silently." is not an option. 

I would agree that if writing such function again, one may come up with a better way, possibly simply not allowing for width/height in inches, therefore being able to use a different default transform altogether etc. The aim of this PR is to restore the old behaviour. Anything else can go outside of `axes_grid1` into whatever new fancy features to come.